### PR TITLE
[DOC] Update code formatting description in `CONTRIBUTING.md`

### PR DESCRIPTION
**Pull request by zhoujinsong**
## Why are the changes needed?

As #1402 replace the code formatting util to spotless, we should update the code formatting description in `CONTRIBUTING.md`

## Brief change log

- Update Code formatting description in `CONTRIBUTING.md`
- Remove useless style configuration files
- Remove useless maven plugins

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (no)
- If yes, how is the feature documented? (not applicable)

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -18,34 +18,43 @@
...
 
 Pull requests are the preferred mechanism for contributing to Amoro
+
 * Generally, create a PR only to the master branch.
 * PR should be linked to the corresponding issue.
-   * The PR title format should be: \[ARCTIC-{issue_number}\]\[{module}\]{pr_description}.
-   * Add fix/resolve #{issue_number} in the description to link the PR to the issue.
-   * The linked issue should clearly explain the background, objectives, and implementation methods of the PR.
-* The change log in the PR should clearly describe the changes made in modules, classes, methods, etc.
+    * The PR title format should be: \[ARCTIC-{issue_number}\]\[{module}\]{pr_description}.
```
Comment (YesOrNo828) [inline CONTRIBUTING.md:44]:
```suggestion
    * The PR title format should be: \[AMORO-{issue_number}\]\[{module}\]{pr_description}.
```

**Review by YesOrNo828 — COMMENTED** [review]
I think we should rename the arctic-checkstle.xml to amoro-checkstyle.xml

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -59,25 +68,55 @@ Code review is a crucial aspect of contributing to a project, and all contributo
 
 ## Design document
 
-Write down your implementation plan and discuss it with other developers in the community before you start coding officially.
-If it is just a small changes, describe the implementation steps clearly in the Issue.
-If it is a relatively large work, it is recommended to write a design document for this feature.
-Here is a [design document template](https://docs.google.com/document/d/1LeTyrlzQJfSs2DkRBsucK_vV5gtHRYLb1KSrpu0hp3g/edit?usp=sharing) for reference.
+Write down your implementation plan and discuss it with other developers in the community before you
+start coding officially. If it is just a small changes, describe the implementation steps clearly in
```
Comment (HuangFru) [inline CONTRIBUTING.md:72]:
```suggestion
start coding officially. If it is just a small change, describe the implementation steps clearly in
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -59,25 +68,55 @@ Code review is a crucial aspect of contributing to a project, and all contributo
...
+7. Under “Formatting Actions”, select “Optimize imports” and “Reformat file”.
+8. From the “All file types list” next to “Reformat code”, select Java and Scala.
 
 ### Copyright
-All files (including source code, configuration files) in the project are required to declare CopyRight information at the top, and the project uses Apache License 2.
-You can cnfigure the copyright information in IntelliJ IDEA with follow steps:
+
+All files (including source code, configuration files) in the project are required to declare
+CopyRight information at the top, and the project uses Apache License 2. You can configure the
+copyright information in IntelliJ IDEA with follow steps:
```
Comment (HuangFru) [inline CONTRIBUTING.md:114]:
```suggestion
copyright information in IntelliJ IDEA with the following steps:
```

**Review by HuangFru — CHANGES_REQUESTED** [review]

**Review by HuangFru — APPROVED** [review]
LGTM.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -59,25 +68,55 @@ Code review is a crucial aspect of contributing to a project, and all contributo
...
 
-Write down your implementation plan and discuss it with other developers in the community before you start coding officially.
-If it is just a small changes, describe the implementation steps clearly in the Issue.
-If it is a relatively large work, it is recommended to write a design document for this feature.
-Here is a [design document template](https://docs.google.com/document/d/1LeTyrlzQJfSs2DkRBsucK_vV5gtHRYLb1KSrpu0hp3g/edit?usp=sharing) for reference.
+Write down your implementation plan and discuss it with other developers in the community before you
+start coding officially. If it is just a small change, describe the implementation steps clearly in
+the Issue. If it is a relatively large work, it is recommended to write a design document for this
+feature. Here is
+a [design document template](https://docs.google.com/document/d/1LeTyrlzQJfSs2DkRBsucK_vV5gtHRYLb1KSrpu0hp3g/edit?usp=sharing)
```
Comment (wangtaohz) [inline CONTRIBUTING.md:75]: The design document template should be rename from `Arctic` to `Amoro`.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -59,25 +68,55 @@ Code review is a crucial aspect of contributing to a project, and all contributo
...
+
+After installing the plugin, format your code automatically by applying the following settings:
+
+1. Go to “Settings/Preferences” → “Other Settings” → “google-java-format Settings”.
+2. Tick the checkbox to enable the plugin.
+3. Change the code style to “Default Google Java style”.
+4. Go to “Settings/Preferences” → Editor → Code Style → Scala.
+5. Change the “Formatter” to “scalafmt”.
+6. Go to “Settings/Preferences” → “Tools” → “Actions on Save”.
+7. Under “Formatting Actions”, select “Optimize imports” and “Reformat file”.
```
Comment (wangtaohz) [inline CONTRIBUTING.md:107]: Should `Reformat file` be `Reformat code` ? (My IntelliJ IDEA 2021.2.3 (Community Edition))
image

File: CONTRIBUTING.md (hunk 4 above)
Comment (zhoujinsong) [reply CONTRIBUTING.md:75]: Thanks for the reminder, I have changed the doc name and translated the doc to the English version.

**Review by wangtaohz — APPROVED** [review]
LGTM. 👍 

I followed the steps to configure my local IDEA, and now automatic formatting and automatic importing are both effective, which is very helpful.

**Pull request merged**